server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/PushSubscriptionSetDeletePerformer.scala

```diff
@@ -0,0 +1,74 @@
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.jmap.method
+
+import org.apache.james.jmap.api.model.{PushSubscriptionId, PushSubscriptionNotFoundException}
+import org.apache.james.jmap.api.pushsubscription.PushSubscriptionRepository
+import org.apache.james.jmap.core.SetError.SetErrorDescription
+import org.apache.james.jmap.core.{PushSubscriptionSetRequest, SetError, UnparsedPushSubscriptionId}
+import org.apache.james.jmap.method.PushSubscriptionSetDeletePerformer.{PushSubscriptionDeletionFailure, PushSubscriptionDeletionResult, PushSubscriptionDeletionResults, PushSubscriptionDeletionSuccess}
+import org.apache.james.mailbox.MailboxSession
+import reactor.core.scala.publisher.{SFlux, SMono}
+import reactor.core.scheduler.Schedulers
+
+import javax.inject.Inject
+
+object PushSubscriptionSetDeletePerformer {
+  sealed trait PushSubscriptionDeletionResult
+  case class PushSubscriptionDeletionSuccess(PushSubscriptionId: PushSubscriptionId) extends PushSubscriptionDeletionResult
+  case class PushSubscriptionDeletionFailure(PushSubscriptionId: UnparsedPushSubscriptionId, exception: Throwable) extends PushSubscriptionDeletionResult {
+    def asPushSubscriptionSetError: SetError = exception match {
+      case e: PushSubscriptionNotFoundException => SetError.notFound(SetErrorDescription(e.getMessage))
+      case e: IllegalArgumentException => SetError.invalidArguments(SetErrorDescription(s"${PushSubscriptionId.id} is not a PushSubscriptionId: ${e.getMessage}"))
+      case _ => SetError.serverFail(SetErrorDescription(exception.getMessage))
+    }
+  }
+  case class PushSubscriptionDeletionResults(results: Seq[PushSubscriptionDeletionResult]) {
+    def destroyed: Seq[PushSubscriptionId] =
+      results.flatMap(result => result match {
+        case success: PushSubscriptionDeletionSuccess => Some(success)
+        case _ => None
+      }).map(_.PushSubscriptionId)
+
+    def retrieveErrors: Map[UnparsedPushSubscriptionId, SetError] =
+      results.flatMap(result => result match {
+        case failure: PushSubscriptionDeletionFailure => Some(failure.PushSubscriptionId, failure.asPushSubscriptionSetError)
+        case _ => None
+      })
+        .toMap
+  }
+}
+
+class PushSubscriptionSetDeletePerformer @Inject()(pushSubscriptionRepository: PushSubscriptionRepository) {
+  def deletePushSubscriptions(pushSubscriptionSetRequest: PushSubscriptionSetRequest, mailboxSession: MailboxSession): SMono[PushSubscriptionDeletionResults] =
+    SFlux.fromIterable(pushSubscriptionSetRequest.destroy.getOrElse(Seq()))
+      .flatMap(unparsedId => delete(unparsedId, mailboxSession)
+        .onErrorRecover(e => PushSubscriptionDeletionFailure(unparsedId, e)),
+        maxConcurrency = 5)
```

Comment:
I remember we have a common const for this

Reply:
Looked in the code by doing a search on `maxConcurrency`, every occurrence is defined like this. Where is that const???

Reply:
We could IMO use ReactorUtils.DEFAULT_CONCURRENCY

Reply:
Should we? ReactorUtils.DEFAULT_CONCURRENCY = 16... not 5.

What is the impact?

Reply:
I'm OK keeping it as it is.